java/src/main/java/com/google/openlocationcode/OpenLocationCode.java

```diff
@@ -665,13 +665,13 @@ private static double clipLatitude(double latitude) {
   }
 
   private static double normalizeLongitude(double longitude) {
-    while (longitude < -LONGITUDE_MAX) {
-      longitude = longitude + LONGITUDE_MAX * 2;
+    if (longitude >= -LONGITUDE_MAX && longitude < LONGITUDE_MAX) {
+      // longitude is within proper range, no normalization necessary
+      return longitude;
     }
-    while (longitude >= LONGITUDE_MAX) {
-      longitude = longitude - LONGITUDE_MAX * 2;
-    }
-    return longitude;
+
+    final long CIRCLE_DEG = 2 * LONGITUDE_MAX; // 360 degrees
+    return (longitude % CIRCLE_DEG + CIRCLE_DEG + LONGITUDE_MAX) % CIRCLE_DEG - LONGITUDE_MAX;
```

Comment:
Can this just be (longitude + LONGITUDE_MAX) % CIRCLE_DEG - LONGITUDE_MAX? If not, please add a comment in the code about why, I got a bit stuck trying to figure it out.

Reply:
This is due to % in JAVA using truncated division, meaning that the remainder can be negative.

With a potentially very large negative input, just adding LONGITUDE_MAX once might not be enough to bring this into positive range, in which case the single % would lead to a still negative result and subtracting LONGITUDE_MAX would bring the value back into an invalid range < -180.

By doing %CIRCLE_DEG twice, we first bring the value into a range that is [-360..360[, so that the second % after adding 360 leads to a result that is definitely in range [0..360[.

I added a comment, please see if it explains this well enough. :)

Reply:
Gotcha, thank you!